compiler/rustc_middle/src/ty/util.rs

```diff
@@ -341,19 +341,19 @@ impl<'tcx> TyCtxt<'tcx> {
     pub fn calculate_dtor(
         self,
         adt_did: DefId,
-        validate: &mut dyn FnMut(Self, DefId) -> Result<(), ErrorReported>,
+        validate: impl Fn(Self, DefId) -> Result<(), ErrorReported>,
```

Comment:
Is there a reason you changed this? This will affect compile times and codegen.

Reply:
I would actually like to know the impact of this line, since only one caller passes in a meaningful closure. In the end, it's a fallout of the no-mut change, so I can undo.

Reply:
If you use `impl`, this will perform codegen for each place `calculate_dtor` is called, which will hurt bootstrap times. Whether it affects runtime is tricky - it does change an indirect call to a direct one, but it also causes more code in the binary which can hurt the instruction cache. I wouldn't change it unless you had a strong reason to.

Reply:
Can we try?

Reply:
If you want to change this, please do it in a separate PR so we can measure the perf more accurately.

Reply:
I'll do that, thanks.